src/PowerShellEditorServices.Engine/Services/Analysis/AnalysisService.cs

```diff
@@ -673,6 +691,232 @@ private class PowerShellResult
 
             public bool HasErrors { get; }
         }
+
+        internal async Task RunScriptDiagnosticsAsync(
+            ScriptFile[] filesToAnalyze)
+        {
+            // If there's an existing task, attempt to cancel it
+            try
+            {
+                if (s_existingRequestCancellation != null)
+                {
+                    // Try to cancel the request
+                    s_existingRequestCancellation.Cancel();
+
+                    // If cancellation didn't throw an exception,
+                    // clean up the existing token
+                    s_existingRequestCancellation.Dispose();
+                    s_existingRequestCancellation = null;
+                }
+            }
+            catch (Exception e)
+            {
+                // TODO: Catch a more specific exception!
+                _logger.LogError(
+                    string.Format(
+                        "Exception while canceling analysis task:\n\n{0}",
+                        e.ToString()));
+
+                TaskCompletionSource<bool> cancelTask = new TaskCompletionSource<bool>();
```

Comment:
What does this bit do? It doesn't seem to hook in to anything other than its own local variable